source/vir/src/prelude.rs

```diff
@@ -523,3 +523,21 @@ pub(crate) fn datatype_height_axiom(typ_name1: &Path, typ_name2: &Path, field: &
         )))
     )
 }
+
+pub(crate) fn datatype_height_axiom_generic(typ_name1: &Path, field: &Ident) -> Node {
```

Comment:
Rather than duplicating `datatype_height_axiom`, let's make `typ_name2` an `Option`.  Then we can splice in the node for either `([box_t2] ([field] x))` or `([field] x)` based on the `Option`.